source_control/git.py

```diff
@@ -469,7 +469,7 @@ def get_head_branch(git_path, module, dest, remote, bare=False):
     if is_not_a_branch(git_path, module, dest):
         f.close()
         f = open(os.path.join(repo_path, 'refs', 'remotes', remote, 'HEAD'))
-    branch = f.readline().split('/')[-1].rstrip("\n")
+    branch = f.readline().split('/',2)[-1].replace(remote,'').lstrip("/").rstrip("\n")
```

Comment:
`.replace(remote, '')` is dangerous, since it replaces all occurrences. What if my branch is named `original-idea` and the remote is named `origin`?  This new code would return a branch name of `al-idea`.